src/hp/netcheck.rs

```diff
@@ -2096,4 +2097,41 @@ mod tests {
 
         Ok(())
     }
+
+    #[tokio::test]
+    async fn test_hairpin() -> Result<()> {
+        setup_logging();
+
+        // Setup STUN server and create derpmap.
```

Comment:
I don't quite get the test setup, could you add a couple of lines of how the flow of packets in here mirrors how hair pinning is supposed to work?